flaky/_flaky_plugin.py

```diff
@@ -127,33 +137,31 @@ def _handle_test_error_or_failure(self, test, err):
             _, _, name = self._get_test_declaration_callable_and_name(test)
         except AttributeError:
             return False
-        current_runs = self._get_flaky_attribute(
-            test,
-            FlakyNames.CURRENT_RUNS,
-        )
-        if current_runs is None:
-            return False
-        current_runs += 1
-        self._set_flaky_attribute(
+
+        need_reruns, should_rerun = self._should_handle_test_error_or_failure(
             test,
-            FlakyNames.CURRENT_RUNS,
-            current_runs,
+            name,
+            err,
         )
-        self._add_flaky_test_failure(test, err)
-        flaky = self._get_flaky_attributes(test)
 
-        if not self._has_flaky_test_failed(flaky):
-            if self._should_rerun_test(test, name, err):
-                self._log_intermediate_failure(err, flaky, name)
-                self._rerun_test(test)
-                return True
+        if self._has_flaky_attributes(test):
+            self._add_flaky_test_failure(test, err)
+            flaky = self._get_flaky_attributes(test)
```

Comment:
Possible refactor: just use _get_flaky_attributes instead of also using _has_flaky_attributes, similar to how we use getattr(, , None) sometimes instead of hasattr()